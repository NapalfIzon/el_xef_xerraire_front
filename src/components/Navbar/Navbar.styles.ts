import { StyleSheet } from "react-native";
import { colors } from "../../styles/colors.styles";

export default StyleSheet.create({
  body: {
    backgroundColor: colors.mainColor,
    flex: 1,
    justifyContent: "center",
    justifyItems: "center",
    alignItems: "center",
  },
});
