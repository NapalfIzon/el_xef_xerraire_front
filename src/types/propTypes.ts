export type RecipeProps = {
  recipes: Array<string>;
};

export type NavigationProps = {
  navigation: {
    navigate: (type: string) => { type: string };
  };
};
