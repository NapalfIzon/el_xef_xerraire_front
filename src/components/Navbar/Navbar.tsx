import React from "react";
import { HStack, IconButton, Icon } from "native-base";
import { FontAwesomeIcon } from "@fortawesome/react-native-fontawesome";
import { NavigationProps } from "../../types/propTypes";
import {
  faHome,
  faSearch,
  faPlusCircle,
} from "@fortawesome/free-solid-svg-icons";
import { faBookmark, faUserCircle } from "@fortawesome/free-regular-svg-icons";

const Navbar = ({ navigation }: NavigationProps) => {
  return (
    <>
      <HStack
        bg="#AB9B7A"
        px="1"
        py="3"
        justifyContent="center"
        justifyItems="center"
        alignItems="center"
      >
        <HStack space="8" alignItems="center" paddingBottom="2">
          <IconButton
            onPress={() => navigation.navigate("Home")}
            icon={
              <Icon
                size="lg"
                as={<FontAwesomeIcon icon={faHome} />}
                color="white"
              />
            }
          />
          <IconButton
            onPress={() => navigation.navigate("Search")}
            icon={
              <Icon
                as={<FontAwesomeIcon icon={faSearch} />}
                size="xl"
                color="white"
              />
            }
          />
          <IconButton
            onPress={() => navigation.navigate("Add your recipe")}
            icon={
              <Icon
                as={<FontAwesomeIcon icon={faPlusCircle} />}
                color="white"
                size="sm"
              />
            }
          />
          <IconButton
            onPress={() => navigation.navigate("Profile")}
            icon={
              <Icon
                as={<FontAwesomeIcon icon={faBookmark} />}
                size="sm"
                color="green"
              />
            }
          />
          <IconButton
            onPress={() => navigation.navigate("Login")}
            icon={
              <Icon
                as={<FontAwesomeIcon icon={faUserCircle} />}
                size="sm"
                color="white"
              />
            }
          />
        </HStack>
      </HStack>
    </>
  );
};

export default Navbar;
