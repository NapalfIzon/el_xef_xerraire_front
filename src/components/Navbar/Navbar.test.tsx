import React from "react";
import { render } from "@testing-library/react-native";
import { NavigationProps } from "../../types/propTypes";
import Navbar from "./Navbar";

jest.mock("native-base", () => ({
  HStack: "",
  IconButton: "",
  Icon: "",
}));
jest.mock("@fortawesome/react-native-fontawesome", () => ({
  FontAwesomeIcon: "",
}));

describe("Given a Navbar component,", () => {
  describe("When it is rendered,", () => {
    test("Then it should render the same as the snapshot.", () => {
      const navigationTest: NavigationProps = {
        navigation: {
          navigate: jest.fn(),
        },
      };

      const result = render(<Navbar navigation={navigationTest} />);

      expect(result).toMatchSnapshot();
    });
  });

  test("Then it should render al the Navbar buttons.", () => {
    const navigationTest: NavigationProps = {
      navigation: {
        navigate: jest.fn(),
      },
    };

    const screen = render(<Navbar navigation={navigationTest} />);
    const buttons = screen.getAllByRole("button");
    const homeButton = screen.getByLabelText("Home");
    const searchButton = screen.getByLabelText("Search");
    const addRecipeButton = screen.getByLabelText("Add your recipe");
    const profileButton = screen.getByLabelText("Profile");
    const loginButton = screen.getByLabelText("Login");

    expect(buttons).not.toBeNull();
    expect(homeButton).not.toBeNull();
    expect(searchButton).not.toBeNull();
    expect(addRecipeButton).not.toBeNull();
    expect(profileButton).not.toBeNull();
    expect(loginButton).not.toBeNull();
  });
});
