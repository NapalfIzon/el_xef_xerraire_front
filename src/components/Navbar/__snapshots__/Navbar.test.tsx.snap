// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a Navbar component, When it is rendered, Then it should render the same as the snapshot. 1`] = `
<
  alignItems="center"
  bg="#AB9B7A"
  justifyContent="center"
  justifyItems="center"
  px="1"
  py="3"
>
  <
    alignItems="center"
    paddingBottom="2"
    space="8"
  >
    <
      accessibilityLabel="Home"
      accessibilityRole="button"
      icon={
        <
          as={
            <
              icon={
                Object {
                  "icon": Array [
                    576,
                    512,
                    Array [],
                    "f015",
                    "M280.37 148.26L96 300.11V464a16 16 0 0 0 16 16l112.06-.29a16 16 0 0 0 15.92-16V368a16 16 0 0 1 16-16h64a16 16 0 0 1 16 16v95.64a16 16 0 0 0 16 16.05L464 480a16 16 0 0 0 16-16V300L295.67 148.26a12.19 12.19 0 0 0-15.3 0zM571.6 251.47L488 182.56V44.05a12 12 0 0 0-12-12h-56a12 12 0 0 0-12 12v72.61L318.47 43a48 48 0 0 0-61 0L4.34 251.47a12 12 0 0 0-1.6 16.9l25.5 31A12 12 0 0 0 45.15 301l235.22-193.74a12.19 12.19 0 0 1 15.3 0L530.9 301a12 12 0 0 0 16.9-1.6l25.5-31a12 12 0 0 0-1.7-16.93z",
                  ],
                  "iconName": "home",
                  "prefix": "fas",
                }
              }
            />
          }
          color="white"
          size="lg"
        />
      }
      onPress={[Function]}
    />
    <
      accessibilityLabel="Search"
      accessibilityRole="button"
      icon={
        <
          as={
            <
              icon={
                Object {
                  "icon": Array [
                    512,
                    512,
                    Array [],
                    "f002",
                    "M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z",
                  ],
                  "iconName": "search",
                  "prefix": "fas",
                }
              }
            />
          }
          color="white"
          size="xl"
        />
      }
      onPress={[Function]}
    />
    <
      accessibilityLabel="Add your recipe"
      accessibilityRole="button"
      icon={
        <
          as={
            <
              icon={
                Object {
                  "icon": Array [
                    512,
                    512,
                    Array [],
                    "f055",
                    "M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm144 276c0 6.6-5.4 12-12 12h-92v92c0 6.6-5.4 12-12 12h-56c-6.6 0-12-5.4-12-12v-92h-92c-6.6 0-12-5.4-12-12v-56c0-6.6 5.4-12 12-12h92v-92c0-6.6 5.4-12 12-12h56c6.6 0 12 5.4 12 12v92h92c6.6 0 12 5.4 12 12v56z",
                  ],
                  "iconName": "plus-circle",
                  "prefix": "fas",
                }
              }
            />
          }
          color="white"
          size="sm"
        />
      }
      onPress={[Function]}
    />
    <
      accessibilityLabel="Profile"
      accessibilityRole="button"
      icon={
        <
          as={
            <
              icon={
                Object {
                  "icon": Array [
                    384,
                    512,
                    Array [],
                    "f02e",
                    "M336 0H48C21.49 0 0 21.49 0 48v464l192-112 192 112V48c0-26.51-21.49-48-48-48zm0 428.43l-144-84-144 84V54a6 6 0 0 1 6-6h276c3.314 0 6 2.683 6 5.996V428.43z",
                  ],
                  "iconName": "bookmark",
                  "prefix": "far",
                }
              }
            />
          }
          color="green"
          size="sm"
        />
      }
      onPress={[Function]}
    />
    <
      accessibilityLabel="Login"
      accessibilityRole="button"
      icon={
        <
          as={
            <
              icon={
                Object {
                  "icon": Array [
                    496,
                    512,
                    Array [],
                    "f2bd",
                    "M248 104c-53 0-96 43-96 96s43 96 96 96 96-43 96-96-43-96-96-96zm0 144c-26.5 0-48-21.5-48-48s21.5-48 48-48 48 21.5 48 48-21.5 48-48 48zm0-240C111 8 0 119 0 256s111 248 248 248 248-111 248-248S385 8 248 8zm0 448c-49.7 0-95.1-18.3-130.1-48.4 14.9-23 40.4-38.6 69.6-39.5 20.8 6.4 40.6 9.6 60.5 9.6s39.7-3.1 60.5-9.6c29.2 1 54.7 16.5 69.6 39.5-35 30.1-80.4 48.4-130.1 48.4zm162.7-84.1c-24.4-31.4-62.1-51.9-105.1-51.9-10.2 0-26 9.6-57.6 9.6-31.5 0-47.4-9.6-57.6-9.6-42.9 0-80.6 20.5-105.1 51.9C61.9 339.2 48 299.2 48 256c0-110.3 89.7-200 200-200s200 89.7 200 200c0 43.2-13.9 83.2-37.3 115.9z",
                  ],
                  "iconName": "user-circle",
                  "prefix": "far",
                }
              }
            />
          }
          color="white"
          size="sm"
        />
      }
      onPress={[Function]}
    />
  </>
</>
`;
